packages/core/admin/admin/src/content-manager/history/components/VersionsList.tsx

```diff
@@ -1,19 +1,259 @@
+import { Contracts } from '@strapi/plugin-content-manager/_internal/shared';
+import { formatDistanceToNowStrict } from 'date-fns';
+import * as locales from 'date-fns/locale';
+import { stringify } from 'qs';
+import { type MessageDescriptor, useIntl } from 'react-intl';
+import { Link, useNavigate } from 'react-router-dom';
+
+import { buildValidGetParams } from '../../utils/api';
+import { useGetHistoryVersionsQuery } from '../services/historyVersion';
+
+import type { Entity, UID } from '@strapi/types';
+
+/* -------------------------------------------------------------------------------------------------
+ * BlueChunks
+ * -----------------------------------------------------------------------------------------------*/
+
+const BlueChunks = (chunks: React.ReactNode) => (
+  <Typography textColor="primary600">{chunks}</Typography>
+);
+
+/* -------------------------------------------------------------------------------------------------
+ * VersionCard
+ * -----------------------------------------------------------------------------------------------*/
+
+type Color = React.ComponentProps<typeof Box>['color'];
+
+interface StatusData {
+  background: Color;
+  border: Color;
+  text: Color;
+  message: MessageDescriptor;
+}
+
+interface VersionCardProps {
+  version: Contracts.HistoryVersions.GetHistoryVersions.Response['data'][number];
+  isCurrent: boolean;
+}
+
+const VersionCard = ({ version, isCurrent }: VersionCardProps) => {
+  const { formatDate, formatMessage, locale } = useIntl();
+  const [{ query }] = useQueryParams<{ id?: string }>();
+
+  const isActive = query.id === version.id.toString();
+
+  const getStatusData = (): StatusData => {
+    switch (version.status) {
+      case 'draft':
+        return {
+          background: 'secondary100',
+          border: 'secondary200',
+          text: 'secondary700',
+          message: {
+            id: 'content-manager.containers.List.draft',
+            defaultMessage: 'Draft',
+          },
+        };
+      case 'modified':
+        return {
+          background: 'alternative100',
+          border: 'alternative200',
+          text: 'alternative700',
+          message: {
+            // TODO: check the translation key once D&P v5 is done
+            id: 'content-manager.containers.List.modified',
+            defaultMessage: 'Modified',
+          },
+        };
+      case 'published':
+      default:
+        return {
+          background: 'success100',
+          border: 'success200',
+          text: 'success700',
+          message: {
+            id: 'content-manager.containers.List.published',
+            defaultMessage: 'Published',
+          },
+        };
+    }
+  };
+
+  const statusData = getStatusData();
+
+  const distanceToNow = formatDistanceToNowStrict(new Date(version.createdAt), {
+    addSuffix: true,
+    // @ts-expect-error - I don't know how to fix this
+    locale: locales[locale],
```

Comment:
whats the error for this? it could be quite straight forward 🤔 is it to do with indexing an object?

Reply:
Yeah for this one I [copied it from there](https://github.com/strapi/strapi/blob/863645fb06a45639020578dab7770bbe45a0e302/packages/core/admin/admin/src/pages/Settings/pages/ApiTokens/EditView/utils/getDateOfExpiration.ts#L13), and I admit I didn't try to fix it after seeing the log.

I updated it to cast the value instead, I think it's slightly better this way. But yes the error is that `locale` from `useIntl` is a plain string type, so `locales[locale]` could be undefined